app/js/enhancedDate.js

```diff
@@ -1,5 +1,144 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+/*
+ * Carolyn Velez
+ * JSCRIPT 300
+ * 04/14/15
+ */
 
+var enhancedDate = (function iife() {
+
+	// private
+	var myDate;
+
+	// public API
+	return {
+
+		setDate: function setDate(specificDate) {
+			if (specificDate === undefined) {
+				myDate = new Date;
+			} else {
+				myDate = specificDate;
+			}
+		},
+
+		getDate: function getDate(dateInMilliseconds) {
+
+			// create variable to store the value of the number of milliseconds
+			var dateMilliseconds;
+
+			/*
+			 compare number of milliseconds between data user entered
+			 and the number of millisconds for today (default)
+			 */
+			if (dateInMilliseconds === true) {
+				return myDate;
+			} else {
+				dateMilliseconds = myDate.getTime();
+				return dateMilliseconds;
+			}
+		},
+
+		getDayName: function getDayName() {
+
+			// create variable to store the value of the week
+			var dayValue;
+
+			// create an array of the names of the days of the week
+			var weekdays = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+
+			// if date user entered in undefined, return today's date
+			if (myDate === undefined) {
```

Comment:
This is good - I'd go one step further and put this check and follow-up step into its own function that each of your methods can use since all of them except setDate should do the same check and follow-up step.